### DISCUSSION TITLE
skupper gateway deprecated/removed?

### DISCUSSION THREAD

**Discussion by markusjnagel** [skupperproject/skupper#1238] [Q&A]
Hi folks,

just a quick question - I know I can build a skupper site (podman) on a VM with a specific IP/Port combination, so that other services/VMs can connect to it and skupper then routes the request to wherever it goes to.

e.g.: `skupper service create <myservicename> port --host-ip a.b.c.d --host-port 8080 ` 

So, all connection requests to `a.b.c.d:8080` inside my network would be "forwarded" via the skupper network of routers, using this router (under IP a.b.c.d ) as a "gateway".

All good - works fine. However, when I wanted to play around with the skupper gateway (which is by the way still part of  `skupper completion` I noticed, it has been removed from the CLI? Or am I just looking in the wrong place - "forest for the trees problem"?

From the "completion":
```
[mnagel@dashboard ~]$ skupper 
completion     delete         gateway        link           revoke-access  status         unexpose       version        
debug          expose         init           network        service        token          update         
```
but then... 

```
[mnagel@dashboard ~]$ skupper gateway --help
Error: unknown command "gateway" for "skupper"
Run 'skupper --help' for usage.
``` 
Because:
```
[mnagel@dashboard ~]$ skupper --help
Usage:
  skupper [command]

Available Commands:
  delete        Delete skupper installation
  expose        Expose a set of pods through a Skupper address
  help          Help about any command
  init          Initialise skupper installation
  link          Manage skupper links definitions
  revoke-access Revoke all previously granted access to the site.
  service       Manage skupper service definitions
  status        Report the status of the current Skupper site
  token         Manage skupper tokens
  unexpose      Unexpose a set of pods previously exposed through a Skupper address
  version       Report the version of the Skupper CLI and services

Flags:
``` 

So, basically all the examples using `gateway` won't work anymore (I know they would, if you simply create a service with a host:port binding - but not verbatim as they are right now? 

For completeness: 
```
[mnagel@dashboard ~]$ skupper version
client version                 1.4.3
transport version              quay.io/skupper/skupper-router:2.4.3 (sha256:2ac1325f0a8a)
[mnagel@dashboard ~]$ cat /etc/os-release 
NAME="Fedora Linux"
VERSION="38 (Server Edition)"
...
```

**Comment by ajssmith** [comment]
Seems to work ok for me with 1.4.3 sans completion

**Comment by omersiar** [comment]
```
./skupper version
client version                 1.7.0
```

For me platform determines if gateway is allowed or not

```
./skupper switch 
podman
```
For podman, gateway does not exist

```
./skupper
Usage:
  skupper [command]

Available Commands:
  delete        Delete skupper installation
  expose        Expose one or more network services
  help          Help about any command
  init          Initialise skupper installation
  link          Manage skupper links definitions
  network       Show information about the sites and services included in the network.
  revoke-access Revoke all previously granted access to the site.
  service       Manage skupper service definitions
  status        Report the status of the current Skupper site
  switch        Select the platform to manage (valid platforms: kubernetes, podman, -)
  token         Manage skupper tokens
  unexpose      Unexpose one or more network services
  update        Update skupper installation version
  version       Report the version of the Skupper CLI and services

Flags:
  -h, --help              help for skupper
      --platform string   The platform type to use [kubernetes, podman]

Use "skupper [command] --help" for more information about a command.
```
